test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ConditionalUpdateTests.cs

```diff
@@ -0,0 +1,182 @@
+
+            Assert.Equal(HttpStatusCode.Created, updateResponse.StatusCode);
+
+            Observation updatedResource = updateResponse.Resource;
+
+            Assert.NotNull(updatedResource);
+            Assert.Equal(observation.Id, updatedResource.Id);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAResourceWithNoId_WhenUpsertingConditionallyWithOneMatch_TheServerShouldReturnTheUpdatedResourceSuccessfully()
+        {
+            var observation = Samples.GetDefaultObservation().ToPoco<Observation>();
+            var identifier = Guid.NewGuid().ToString();
+
+            observation.Identifier.Add(new Identifier("http://e2etests", identifier));
+            FhirResponse<Observation> response = await Client.CreateAsync(observation);
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+            var observation2 = Samples.GetDefaultObservation().ToPoco<Observation>();
+            observation2.Id = null;
+            observation2.Identifier.Add(new Identifier("http://e2etests", identifier));
+            observation2.Text.Div = "<div>Updated!</div>";
+            FhirResponse<Observation> updateResponse = await Client.ConditionalUpdateAsync(
+                observation2,
+                $"identifier={identifier}");
+
+            Assert.Equal(HttpStatusCode.OK, updateResponse.StatusCode);
+
+            Observation updatedResource = updateResponse.Resource;
+
+            Assert.NotNull(updatedResource);
+            Assert.Equal(response.Resource.Id, updatedResource.Id);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenAResourceWithCorrectId_WhenUpsertingConditionallyWithOneMatch_TheServerShouldReturnTheUpdatedResourceSuccessfully()
+        {
+            var observation = Samples.GetDefaultObservation().ToPoco<Observation>();
+            var identifier = Guid.NewGuid().ToString();
+
+            observation.Identifier.Add(new Identifier("http://e2etests", identifier));
+            FhirResponse<Observation> response = await Client.CreateAsync(observation);
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+
+            var observation2 = Samples.GetDefaultObservation().ToPoco<Observation>();
+            observation2.Id = response.Resource.Id;
+            observation2.Identifier.Add(new Identifier("http://e2etests", identifier));
+            observation2.Text.Div = "<div>Updated!</div>";
+            FhirResponse<Observation> updateResponse = await Client.ConditionalUpdateAsync(
+                observation2,
+                $"identifier={identifier}");
+
+            Assert.Equal(HttpStatusCode.OK, updateResponse.StatusCode);
+
+            Observation updatedResource = updateResponse.Resource;
+
+            Assert.NotNull(updatedResource);
+            Assert.Equal(response.Resource.Id, updatedResource.Id);
```

Comment:
Should we also check whether the updated resource has the new Text element? (here and in the previous test)